Compilation breaks with `CONFIG_POSIX_SEMAPHORES=y`

### Describe the bug

Adding `CONFIG_POSIX_SEMAPHORES=y` for POSIX semaphore support without further POSIX Kconfig options can lead to a build breakage like:
```
In file included from <path>/zephyr/lib/posix/options/semaphore.c:8:
<path>/zephyr/lib/posix/options/posix_clock.h:62:32: error: unknown type name 'clockid_t'
   62 | uint32_t timespec_to_timeoutms(clockid_t clock_id, const struct timespec *abstime);
      |                                ^~~~~~~~~
<path>/zephyr/lib/posix/options/posix_clock.h:64:38: error: unknown type name 'clockid_t'
   64 | __syscall int __posix_clock_get_base(clockid_t clock_id, struct timespec *ts);
      |                                      ^~~~~~~~~
<path>/zephyr/lib/posix/options/posix_clock.h:68:10: fatal error: zephyr/syscalls/posix_clock.h: No such file or directory
   68 | #include <zephyr/syscalls/posix_clock.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[145/474] Building C object zephyr/subsys/llext/CMakeFiles/subsys__llext.dir/llext_load.c.obj
ninja: build stopped: subcommand failed.
```
#92457 fixes it

### To reproduce

Observed while building SOF with a `sys_sem_*()` API addition. Interestingly I've so far been unable to reproduce the breakage with Zephyr tests / samples. It seems like with Zephyr builds it's enough that posix_types.h is **indirectly** included **after** posix_clock.h, while with the SOF build it breaks. Still headers should be included where their types are used, and since posix_clock.h uses `clockid_t` posix_types.h should be included there.
**Update:** I've figured out why the SOF build fails while Zephyr test builds succeed: SOF doesn't use the `-specs=...` compiler flag.
**Update 2:** Confirmed, that using `CONFIG_MINIMAL_LIBC=y` reproduces the build breakage

### Impact

Inability to build Zephyr

### Environment

* Commit SHA or Version used:
  - Originally detected with 8c6b2c8c88773810d64a47ed3b6206b7d6e8290f
  - With 4852719f98b3bac9c05196d9855d897f47e3c82f the "missing header" error is gone, but the "unknown type" error stays

--- a/lib/posix/options/posix_clock.h
+++ b/lib/posix/options/posix_clock.h
@@ -9,13 +9,11 @@
 #define ZEPHYR_LIB_POSIX_POSIX_CLOCK_H_
 
 #include <stdbool.h>
-#include <stddef.h>
 #include <stdint.h>
 #include <time.h>
 
-#include <zephyr/sys_clock.h>
-#include <zephyr/sys/__assert.h>
 #include <zephyr/posix/sys/time.h>
+#include <zephyr/sys/clock.h>
 #include <zephyr/sys/timeutil.h>
 
 #ifdef __cplusplus
@@ -59,7 +57,7 @@ static inline bool tp_diff_in_range_ns(const struct timespec *a, const struct ti
 	return diff >= lo && diff < hi;
 }
 
-uint32_t timespec_to_timeoutms(clockid_t clock_id, const struct timespec *abstime);
+uint32_t timespec_to_timeoutms(int clock_id, const struct timespec *abstime);
 
 /** INTERNAL_HIDDEN @endcond */
 

--- a/lib/posix/options/timespec_to_timeout.c
+++ b/lib/posix/options/timespec_to_timeout.c
@@ -8,15 +8,16 @@
 
 #include <limits.h>
 #include <stdint.h>
+#include <time.h>
 
-#include <zephyr/posix/time.h>
+#include <zephyr/sys/clock.h>
 #include <zephyr/sys/util.h>
 
-uint32_t timespec_to_timeoutms(clockid_t clock_id, const struct timespec *abstime)
+uint32_t timespec_to_timeoutms(int clock_id, const struct timespec *abstime)
 {
 	struct timespec curtime;
 
-	if (clock_gettime(clock_id, &curtime) < 0) {
+	if (sys_clock_gettime(clock_id, &curtime) < 0) {
 		return 0;
 	}
 

